Include embark-org in embark package

oantolin opened the pull request:
This PR is a simple change to the MELPA recipe to include the file `embark-org.el` in the embark package.

### Brief summary of what the package does

Embark provides a sort of right-click or contextual menu for Emacs, making it easy to run relevant actions on the current target, which can be a minibuffer completion candidate or the thing at point. The actions offered depend on the type of target, for example, if you are currently in find-file, Embark will offer you actions such as copy-file, delete-file, make-directory, etc. The determination of what the target is and which actions to offer is highly configurable.

This is similar to the notion of actions in the well-known Helm and Ivy packages (though those are limited to acting on minibuffer completion candidates).

### Direct link to the package repository

https://github.com/oantolin/embark

### Your association with the package

I am the author and maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thank you - happy to merge this recipe update, but I'll post some (selected) lints while I'm here:

`embark-org.el` with package-lint 20221003.1636:
```
1 issue found:
152:18: error: You should depend on (emacs "27.1") if you need `org-table-move-cell-up'.
```
(either that, or depend on a specific version of org)

`embark-org.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- embark-org.el#L3: Have you done the paperwork to assign this copyright? (if not, the copyright is yours)```
```

`embark.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- embark.el#L3: Have you done the paperwork to assign this copyright?
- embark.el#L12: This may be a copy-paste error?  Is this in fact a builtin part of Emacs?
- embark.el#L1819: It's safer to sharp-quote function names; use `#'`
- embark.el#L3821: It's safer to sharp-quote function names; use `#'`
- embark.el#L3822: It's safer to sharp-quote function names; use `#'`
- embark.el#L2570: This double negation can be collapsed (`not` aliases `null`)
- embark.el#L2144: Use `when ...` instead of `unless (null ...)` (the containing block looks like it could be rewritten for clarity using an `if` though)
```

oantolin commented:
Hi @riscy! Thank you very much for merging so quickly.

That melpazoid tool looks great!

I have signed the copyright assignment papers for the FSF. As for:

> This may be a copy-paste error?  Is this in fact a builtin part of Emacs?

I was under the impression that packages distributed on GNU ELPA *are* considered part of Emacs. Several other non-built-in packages that are on GNU ELPA have this same comment about being part of Emacs.

I'll adress the other lints, thanks again.

oantolin commented:
I added the dependency on Emacs 27.1 to embark-org, and simplified the double negations in embark. I didn't sharp quote the ivy function, because I want the code to byte-compile cleanly even for users that don't have Ivy installed. Finally, I didn't rewrite that last block as an if because the logic is a little to complicated for an if: it's a series of nested tests some of which introduce bindings, some with when-let, and some with pcase-let, and if any one of the tests in the sequence fails there is a single fallback to jump to.
